File: Sharebook/Server/src/Services/Post/Sharebook.Post.API/DTO/ObjectEntitiesDTO.cs
namespace Sharebook.Post.DTO
{
    using System.Collections.Generic;
    using Newtonsoft.Json;

    using Sharebook.Post.Domain.AggregatesModel.PostAggregate.Entities;

    public class ObjectEntitiesDTO : IObjectEntities
    {
        [JsonProperty("urls")]
        public List<IUrlEntity> Urls { get; set; }

        [JsonProperty("user_mentions")]
        public List<IUserMentionEntity> UserMentions { get; set; }

        [JsonProperty("hashtags")]
        public List<IHashtagEntity> Hashtags { get; set; }

        [JsonProperty("symbols")]
        public List<ISymbolEntity> Symbols { get; set; }

        [JsonProperty("media")]
        public List<IMediaEntity> Medias { get; set; }
    }
}

File: Sharebook/Client/src/app/core/Core/DTO/ObjectEntitiesDTO.ts
import {IObjectEntities} from "../../Public/Models/Entities/IObjectEntities";
import {IUrlEntity} from "../../Public/Models/Entities/IUrlEntity";
import {IUserMentionEntity} from "../../Public/Models/Entities/IUserMentionEntity";
import {IHashtagEntity} from "../../Public/Models/Entities/IHashTagEntity";
import {ISymbolEntity} from "../../Public/Models/Entities/ISymbolEntity";
import {IMediaEntity} from "../../Public/Models/Entities/IMediaEntity";

export class ObjectEntitiesDTO implements IObjectEntities {
  // [JsonProperty("urls")]
  public urls: Array<IUrlEntity>;

  // [JsonProperty("user_mentions")]
  public userMentions: Array<IUserMentionEntity>;

  // [JsonProperty("hashtags")]
  public hashtags: Array<IHashtagEntity>;

  // [JsonProperty("symbols")]
  public symbols: Array<ISymbolEntity>;

  // [JsonProperty("media")]
  public medias: Array<IMediaEntity>;
}
